snapcraft/application.py

```diff
@@ -111,13 +113,23 @@ def _resolve_project_path(self, project_dir: pathlib.Path | None) -> pathlib.Pat
         if project_dir is None:
             project_dir = pathlib.Path.cwd()
 
-        try:
+        with suppress(FileNotFoundError):
             return super()._resolve_project_path(project_dir / "snap")
-        except FileNotFoundError:
-            try:
-                return super()._resolve_project_path(project_dir)
-            except FileNotFoundError:
-                return super()._resolve_project_path(project_dir / "build-aux" / "snap")
+
+        with suppress(FileNotFoundError):
+            return super()._resolve_project_path(project_dir)
+
+        with suppress(FileNotFoundError):
+            return super()._resolve_project_path(project_dir / "build-aux" / "snap")
+
+        raise ProjectFileMissingError(
+            f"Project file '{self.app.name+'.yaml'}' not found in '{project_dir}'",
```

Comment:
The lack of coverage here tells me that we should have at least one test that checks that this message is correctly raised 
also we can use "snapcraft.yaml" in this case here